src/provider/provider_tracking.c

```diff
@@ -147,10 +147,6 @@ static umf_result_t trackingAlloc(void *hProvider, size_t size,
         (umf_tracking_memory_provider_t *)hProvider;
     umf_result_t ret = UMF_RESULT_SUCCESS;
 
-    if (!p->hUpstream) {
```

Comment:
why do you remove that?

Reply:
Because this is checked once at the beginning in `trackingInitialize()`. We can have an assert here instead.

Reply:
assert added instead

Reply:
ok looks good for me